### ISSUE TITLE
Using zsh extansions to customize font type - Color issues

### ISSUE THREAD

**Issue by Sherulez** [romkatv/powerlevel10k#502]
Hi,

I'm trying to make the current time segment (text) _italic_ or **bold**.
I tried in different ways but each time I use zsh extansions I have a color problem.

**Bold, but black :**
```
typeset -g POWERLEVEL9K_TIME_FOREGROUND=red
typeset -g POWERLEVEL9K_TIME_CONTENT_EXPANSION='%B${P9K_CONTENT}'
```
![Capture](https://user-images.githubusercontent.com/25638207/74603693-93abc800-50b6-11ea-88d0-880398d47764.PNG)
I must not do it the right way, using P9K_CONTENT does not seem to be adequate.

-----------------------

**Same with a custom segment :** 
```
function prompt_test() {  
    # p10k segment -f blue -t $'%{\e[3m%}%*%{\e[0m%}' # WORKS - italic escape chars
    p10k segment -f red -t $'%B%*%b'
}
```
![Capture3](https://user-images.githubusercontent.com/25638207/74603855-6102cf00-50b8-11ea-9c3f-07f9ede1c82c.PNG)

-----------------------

**Works well :**
``` 
typeset -g POWERLEVEL9K_TIME_CONTENT_EXPANSION=$'%{\e[3m%} ${P9K_CONTENT} %{\e[0m%}'
```
![Capture2](https://user-images.githubusercontent.com/25638207/74603815-b094cb00-50b7-11ea-900e-dce5101acf95.PNG)

In order for zsh extensions to work properly I have to add `%f`

> %F (%f)
> Start (stop) using a different foreground colour

before, and define the color by hand ... making `POWERLEVEL9K_TIME_FOREGROUND=red` useless.

```
typeset -g POWERLEVEL9K_TIME_CONTENT_EXPANSION='%f%F{red}%B${P9K_CONTENT}%f'
```

**Comment by romkatv** [comment]
What's the output of `echo $ZSH_VERSION`? Is it perchance 5.2?

**Comment by Sherulez** [comment]
5.7.1

**Comment by romkatv** [comment]
OK, let's try to reproduce this. Please run the following command and post a screenshot of the output.

```zsh
() (
  emulate -L zsh -o err_return
  local tmp
  tmp="$(mktemp -d ${TMPDIR:-/tmp}/p10k.XXXXXXXXXX)"
  {
    cd $tmp
    git clone --depth=1 https://github.com/romkatv/powerlevel10k.git
    PROMPT= zsh --no-promptcr -dfis <<\END
POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(time)
POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=()
POWERLEVEL9K_TIME_FOREGROUND=red
POWERLEVEL9K_TIME_CONTENT_EXPANSION='%B${P9K_CONTENT}'
typeset -pm 'ZSH_VERSION|TERM|COLORTERM'
source powerlevel10k/powerlevel10k.zsh-theme
END
    print
  } always {
    cd /
    rm -rf $tmp
  }
)
```

**Comment by Sherulez** [comment]
Well, bold and red ... so the issue is from my conf I guess
![image](https://user-images.githubusercontent.com/25638207/74606092-d6c56580-50cd-11ea-8efa-7cfcc8f6f019.png)

**Comment by romkatv** [comment]
I can debug it if you can figure out how to reproduce.

**Comment by Sherulez** [comment]
Also a strange thing, if the last command status failed, the time turn red, eeeh ?
![Capture5](https://user-images.githubusercontent.com/25638207/74606122-0e341200-50ce-11ea-8fda-649f250de7a3.PNG)

**Comment by Sherulez** [comment]
I will start fresh, with a reinstall of p10k and zsh

**Comment by romkatv** [comment]
Can you try updating powerlevel10k and restarting zsh?

> I will start fresh, with a reinstall of p10k and zsh

This sounds like an overkill.

**Comment by Sherulez** [comment]
I did :

- `git -C $ZSH_CUSTOM/themes/powerlevel10k pull`
- `p10k configure`
- Computer restart
- Added ` typeset -g POWERLEVEL9K_TIME_CONTENT_EXPANSION='%B${P9K_CONTENT}'` below `typeset -g POWERLEVEL9K_TIME_UPDATE_ON_COMMAND=false`

![Capture7](https://user-images.githubusercontent.com/25638207/74606290-8222ea00-50cf-11ea-9115-4128bd2b7266.PNG)

**Comment by Sherulez** [comment]
Oh my bad my bad .... I found the culprit.
I had this old line in my .zshrc

`POWERLEVEL9K_STATUS_OK_FOREGROUND=#47D4B9`

Sorry for wasting your time

**Comment by romkatv** [comment]
This is a bug in Zsh. Didn't know this one.

```zsh
emulate zsh -o prompt_percent -c 'print -P "%F{#ff0000}red%F{green}%B bold green"'
```

**Comment by Sherulez** [comment]
> This is a bug in Zsh. Didn't know this one.
> 
> ```shell
> emulate zsh -o prompt_percent -c 'print -P "%F{#ff0000}red%F{green}%B bold green"'
> ```

Good catch ! the green is black o_o

**Comment by Sherulez** [comment]
this does work : `emulate zsh -o prompt_percent -c 'print -P "%F{#ff0000}red%f%F{green}%B bold green"'`

**Comment by romkatv** [comment]
> this does work : `emulate zsh -o prompt_percent -c 'print -P "%F{#ff0000}red%f%F{green}%B bold green"'`

Yep. Swapping the order of `%F{green}` and `%B` also works.

Closing as I don't feel like adding workarounds for this bug in Powerlevel10k. Affected users can add workarounds themselves. Since triggering this bug requires embedding percent-escapes in `EXPANSION` parameters, users can embed more escapes there.

**Comment by Sherulez** [comment]
> Closing as I don't feel like adding workarounds for this bug in Powerlevel10k. Affected users can add workarounds themselves. Since triggering this bug requires embedding percent-escapes in EXPANSION parameters, users can embed more escapes there.

That make sense.
I take care of opening an issue on this subject on the zsh github or I let you take care of it? :)

**Comment by romkatv** [comment]
If you want to report this bug, you need to send an email to zsh-workers@zsh.org. Use plain text format. If you attach a patch with a fix, that would be awesome. Without a patch it's not very useful.

If this bug was discovered a week ago, I would go and fix it right away in order to get the fix released in zsh 5.8. Unfortunately, zsh 5.8 shipped yesterday.

I've added a TODO for myself to fix the bug when I'm through with more urgent matters.

**Comment by Sherulez** [comment]
> If you attach a patch with a fix, that would be awesome
I will try if my abilities allow :p 

Otherwise I will wait until your TODO transforms into DONE when you have more free time :) Thank you for your support throughout this problem.
